vectorstores/cloudsql/vectorstore.go

```diff
@@ -39,3 +41,72 @@ type SearchDocument struct {
 }
 
 var _ vectorstores.VectorStore = &VectorStore{}
+
+// NewVectorStore creates a new VectorStore with options.
+func NewVectorStore(ctx context.Context, engine cloudsqlutil.PostgresEngine, embedder embeddings.Embedder, tableName string, opts ...CloudSQLVectoreStoresOption) (VectorStore, error) {
+	vs, err := applyCloudSQLVectorStoreOptions(engine, embedder, tableName, opts...)
+	if err != nil {
+		return VectorStore{}, err
+	}
+	return vs, nil
+}
+
+// ApplyVectorIndex creates an index in the table of the embeddings
+func (vs *VectorStore) ApplyVectorIndex(ctx context.Context, index BaseIndex, name string, concurrently, overwrite bool) error {
+	if index.indexType == "exactnearestneighbor" {
```

Comment:
Shouldn't we always drop the index when a new one is applied? 

(Or never, it seems weird to do it just for ANN)

Reply:
on the PYthon version its only when its "exactnearestneighbor" https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/main/src/langchain_google_cloud_sql_pg/async_vectorstore.py#L698 ! I believe that's the way if we are doing a mirror of the python library!